index.js

```diff
@@ -180,2669 +180,2588 @@
-            ),
-            tuples
-          )
-        ),
-      },
-    }),
-    {
-      arity,  // number of type parameters
-      keys: tuples.map (([k]) => k),
-      name,
-      supertypes,
-      type,
-      types: tuples.reduce ((types, [k, , t]) => ((types[k] = t, types)), {}),
-      url,
-    }
-  );
+const production = globalThis.process?.env?.NODE_ENV === 'production';
 
-  const BINARY        = 'BINARY';
-  const FUNCTION      = 'FUNCTION';
-  const INCONSISTENT  = 'INCONSISTENT';
-  const NO_ARGUMENTS  = 'NO_ARGUMENTS';
-  const NULLARY       = 'NULLARY';
-  const RECORD        = 'RECORD';
-  const UNARY         = 'UNARY';
-  const UNKNOWN       = 'UNKNOWN';
-  const VARIABLE      = 'VARIABLE';
-
-  //  Inconsistent :: Type
-  const Inconsistent = _Type (
-    INCONSISTENT,
-    '',
-    '',
-    0,
-    (outer, inner) => '???',
-    [],
-    null,
-    []
-  );
+const env = [];
 
-  //  NoArguments :: Type
-  const NoArguments = _Type (
-    NO_ARGUMENTS,
-    '',
-    '',
-    0,
-    (outer, inner) => '()',
-    [],
-    null,
-    []
-  );
+const def = mkdef ({checkTypes: !production, env});
 
-  //  functionUrl :: String -> String
-  const functionUrl = name => {
-    const version = '0.22.0';  // updated programmatically
-    return (
-      `https://github.com/sanctuary-js/sanctuary-def/tree/v${version}#${name}`
-    );
-  };
+//. ### Types
+//.
+//. Conceptually, a type is a set of values. One can think of a value of
+//. type `Type` as a function of type `Any -> Boolean` that tests values
+//. for membership in the set (though this is an oversimplification).
 
-  const NullaryTypeWithUrl = Z.ap (NullaryType, functionUrl);
-  const UnaryTypeWithUrl = Z.ap (UnaryType, functionUrl);
-  const BinaryTypeWithUrl = Z.ap (BinaryType, functionUrl);
-
-  //. ### Types
-  //.
-  //. Conceptually, a type is a set of values. One can think of a value of
-  //. type `Type` as a function of type `Any -> Boolean` that tests values
-  //. for membership in the set (though this is an oversimplification).
-
-  //# Unknown :: Type
-  //.
-  //. Type used to represent missing type information. The type of `[]`,
-  //. for example, is `Array ???`.
-  //.
-  //. May be used with type constructors when defining environments. Given a
-  //. type constructor `List :: Type -> Type`, one could use `List ($.Unknown)`
-  //. to include an infinite number of types in an environment:
-  //.
-  //.   - `List Number`
-  //.   - `List String`
-  //.   - `List (List Number)`
-  //.   - `List (List String)`
-  //.   - `List (List (List Number))`
-  //.   - `List (List (List String))`
-  //.   - `...`
-  const Unknown = _Type (
-    UNKNOWN,
-    '',
-    '',
-    0,
-    (outer, inner) => 'Unknown',
-    [],
-    env => x => true,
-    []
-  );
+//# Type :: Type
+//.
+//. Type comprising every `Type` value.
+export const Type = NullaryTypeWithUrl
+  ('Type')
+  ([])
+  (x => type (x) === 'sanctuary-def/Type@1');
```

Comment:
I moved this block to the top of the section as other type definitions depend on `Type`.